lib/api.js

```diff
@@ -288,21 +324,131 @@ API.prototype.export = function(opts) {
 API.prototype.import = function(str, opts) {
   opts = opts || {};
 
-  var credentials;
+
+  // Json input?
   try {
+    var credentials;
     if (opts.compressed) {
       credentials = Credentials.importCompressed(str, opts.password);
-      // HACK: simulate incomplete credentials
-      delete credentials.m;
     } else {
       credentials = Credentials.fromObj(JSON.parse(str));
     }
+    this.credentials = credentials;
   } catch (ex) {
-    throw new Error('Error importing from source:' + ex);
+    throw Errors.INVALID_BACKUP;
   }
-  this.credentials = credentials;
 };
 
+
+API.prototype.importFromMnemonic = function(words, opts, cb) {
+  log.info('Importing from 12 Words');
+
+  var nrWords = words.split(' ').map(function(v){
+    return _.trim(v);
+  }).length;
+  if (nrWords != 12)
+    return cb(Errors.INVALID_BACKUP);
+
+  var key = Credentials.keyFromMnemonic(words, opts.passphrase, opts.network);
```

Comment:
Since the parent function is async, any exception thrown here should be returned using the callback